lib/model/src/main/java/io/atlasmap/v2/Document.java

```diff
@@ -54,4 +59,62 @@ public void setFields(Fields value) {
         this.fields = value;
     }
 
+    /**
+     * Exclude fields which are not on requested paths.
+     *
+     * If includePaths is null or empty it does nothing and all fields are included.
+     *
+     * @param includePaths
+     */
+    public void excludeNotRequestedFields(List<String> includePaths) {
```

Comment:
This is better to be in atlas-core, maybe as a static util method or something (e.g. io.atlasmap.core.InspectionUtil next to AtlasUtil). The model objects are better to be only with properties getter/setter and minimize the logic.

Reply:
Also, I wonder if we can smoothly transition when we make backend side inspection on-demand as a 2nd phase. I expect the complex fields are marked with something like `FieldStatus.DEFERRED` and have no children. Will this method be just removed at the 2nd phase?

Reply:
And can we get a unit test once it's relocated?

Reply:
I've moved it to AtlasUtil and unit tested it. I didn't assign FieldStatus.DEFERRED. It would require rewriting the method as I would need to mark parent as deferred and currently I don't pass parent around... just the collection of children. If you think it's important to mark parent as FieldStatus.DEFERRED then I can try to do that.

Reply:
@igarashitm how about this one? Is it okay to not set FieldStatus.DEFERRED? It's a bit more work and UI doesn't need that. UI just checks if there are children loaded for a complex type and if not then it requests them.

Reply:
Sounds good, FieldStatus.DEFERRED is not necessary if it doesn't make something better.